File: czv-python/tests/test_count.py
import czv
import pytest
from .test_data import test_data

class TestRowCount:
    @pytest.mark.parametrize(
        "file_name,expected",
        [("fruits.csv", 3), ("constituents_altnames.csv", 33971)],
    )
    def test_row_count(self, file_name, expected):
        """Count the total number of non-header rows."""

        result = czv.row_count(file_path=test_data[file_name])
        assert result == expected

    @pytest.mark.parametrize(
        "file_name,expected",
        [("fruits.csv", 4), ("constituents_altnames.csv", 33972)],
    )
    def test_row_count_include_header_row(self, file_name, expected):
        """Count the total number of rows including the header row."""

        result = czv.row_count(file_path=test_data[file_name], include_header_row=True)
        assert result == expected

class TestColumnCount:
    @pytest.mark.parametrize(
        "file_name,expected",
        [("fruits.csv", 2), ("constituents_altnames.csv", 6)],
    )
    def test_column_count(self, file_name, expected):
        """Count the total number of columns."""

        result = czv.column_count(file_path=test_data[file_name])
        assert result == expected

File: czv/tests/test_count.rs
use czv;
use czv::Result;

#[test]
fn test_row_count() -> Result<()> {
    let cases = vec![
        ("tests/resources/fruits.csv", 3),
        ("tests/resources/constituents_altnames.csv", 33971),
    ];
    for (file_name, expected) in cases {
        let got = czv::count::row_count(Some(file_name.into()), None, false)?;
        assert_eq!(expected, got);
    }
    Ok(())
}

#[test]
fn test_row_count_builder() -> Result<()> {
    let cases = vec![
        ("tests/resources/fruits.csv", 3),
        ("tests/resources/constituents_altnames.csv", 33971),
    ];
    for (file_name, expected) in cases {
        let got = czv::count::RowCount::new().file_path(file_name).execute()?;
        assert_eq!(expected, got);
    }
    Ok(())
}

#[test]
fn test_column_count() -> Result<()> {
    let cases = vec![
        ("tests/resources/fruits.csv", 2),
        ("tests/resources/constituents_altnames.csv", 6),
    ];
    for (file_name, expected) in cases {
        let got = czv::count::column_count(Some(file_name.into()), None)?;
        assert_eq!(expected, got);
    }
    Ok(())
}

#[test]
fn test_column_count_builder() -> Result<()> {
    let cases = vec![
        ("tests/resources/fruits.csv", 2),
        ("tests/resources/constituents_altnames.csv", 6),
    ];
    for (file_name, expected) in cases {
        let got = czv::count::ColumnCount::new()
            .file_path(file_name)
            .execute()?;
        assert_eq!(expected, got);
    }
    Ok(())
}
